astropy/time/core.py

```diff
@@ -2276,6 +2284,72 @@ def __add__(self, other):
     def __radd__(self, other):
         return self.__add__(other)
 
+    def mean(
+            self: 'Time',
+            axis: Optional[Union[int, Tuple[int]]] = None,
+            dtype: 'npt.DTypeLike' = None,
+            out: Optional['Time'] = None,
+            keepdims: bool = False,
+            *,
+            where: 'npt.NDArray[bool]' = True,
+    ) -> 'Time':
```

Comment:
```suggestion
    ) -> Time:
```

Do we subclass time? Then we need the ``Self`` from ``typing_extensions``. Which Astropy doesn't have as a dependency so would need to be in ``TYPE_CHECKING``.

Reply:
I think it would be nice if we left the option open to subclass `Time`, although maybe there are already other reasons why `Time` can't be subclassed.

Reply:
Note that there is a `TimeBase`, which is subclassed by `Time` and `TimeDelta`. I hadn't realized it, but you're adding to `Time`, while you might as well use `TimeBase`, so `TimeDelta` gets the method too (it should not be any different). In that case, there will be subclasses.

Reply:
Sounds good, I will move it as soon as I resolve all the current conversations.